test/physics/em/SeltzerBerger.test.cc

```diff
@@ -106,3 +130,88 @@ TEST_F(SeltzerBergerTest, sb_tables)
 }
+
+TEST_F(SeltzerBergerTest, sb_positron_xs_scaling)
+{
+    const ParticleParams& pp        = *this->particle_params();
+    const MevMass     positron_mass = pp.get(pp.find(pdg::positron())).mass();
+    const MevEnergy   gamma_cutoff{0.01};
+    const ElementView el = this->material_params()->get(ElementId{0});
+
+    std::vector<real_type> scaling_frac;
+
+    for (real_type inc_energy : {1.0, 10.0, 100., 1000.})
+    {
+        SBPositronXsCorrector scale_xs(
+            positron_mass, el, gamma_cutoff, MevEnergy{inc_energy});
+        for (real_type sampled_efrac : {.10001, .5, .9, .9999})
+        {
+            real_type exit_energy = sampled_efrac * inc_energy;
+            scaling_frac.push_back(scale_xs(MevEnergy{exit_energy}));
+        }
+    }
+    // clang-format off
+    const double expected_scaling_frac[] = {
+        0.9980342329193, 0.9800123976959, 0.8513338610609, 2.827839668085e-05,
+        0.9999449017036, 0.9993355880532, 0.9870854387438, 0.04176283721387,
+        0.9999993627271, 0.9999919054025, 0.9997522383667, 0.4174036918268,
+        0.9999999935293, 0.9999999173093, 0.9999972926228, 0.8399650995661};
+    // clang-format on
+    EXPECT_VEC_SOFT_EQ(expected_scaling_frac, scaling_frac);
+}
+
+TEST_F(SeltzerBergerTest, sb_energy_dist)
+{
+    RandomEngine&   rng_engine = this->rng();
+    const MevEnergy gamma_cutoff{0.0009};
+
+    const int           num_samples = 8192;
+    std::vector<double> max_xs;
+    std::vector<double> avg_exit_frac;
+    std::vector<double> avg_engine_samples;
+
+    // Note: the first point has a very low cross section compared to
+    // ionization so won't be encountered in practice. The differential cross
+    // section distribution is much flatter there, so there should be lower
+    // rejection. The second point is where the maximum of the differential SB
+    // data switches between a high-exit-energy peak and a low-exit-energy
+    // peak, which should result in a higher rejection rate. The remaining
+    // points are arbitrary.
+    for (real_type inc_energy : {0.001, 0.0045, 0.567, 7.89, 89.0, 901.})
+    {
+        SBEnergyDistribution sample_energy(
+            model_->host_pointers(),
+            Energy{inc_energy},
+            ElementId{0},
+            this->density_correction(MaterialId{0}, Energy{inc_energy}),
+            gamma_cutoff);
+        max_xs.push_back(sample_energy.max_xs());
+        double total_exit_energy = 0;
+
+        // Loop over many particles
+        for (int i = 0; i < num_samples; ++i)
+        {
+            Energy exit_gamma = sample_energy(rng_engine);
+            EXPECT_GT(exit_gamma.value(), gamma_cutoff.value());
+            EXPECT_LT(exit_gamma.value(), inc_energy);
+            total_exit_energy += exit_gamma.value();
+        }
+
+        avg_exit_frac.push_back(total_exit_energy / (num_samples * inc_energy));
+        avg_engine_samples.push_back(double(rng_engine.count()) / num_samples);
+        rng_engine.reset_count();
```

Comment:
is the rng counting part of this test?

Reply:
Yep. It too helped me catch a bug in my rejection sampling 😄